From faef6667fb7f0b55978b3019b442fdc0b37de03e Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Thu, 24 Dec 2015 16:36:51 +0800
Subject: [PATCH 1262/1691] MLK-12075 ARM: dts: remove imx7d-sdb-touch.dts

commit faef6667fb7f0b55978b3019b442fdc0b37de03e from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

For imx7d-sdb TO1.1 board, the touch interrupt pin no
longer conflict with HDMI, so move touch support to
imx7d-sdb.dts. To compatible with TO1.0 board, still
use imx7d-sdb-reva-touch.dts to support touch.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
---
 arch/arm/boot/dts/Makefile                 |  3 +--
 arch/arm/boot/dts/imx7d-sdb-reva-touch.dts | 14 ++++++++++-
 arch/arm/boot/dts/imx7d-sdb-reva.dts       |  4 ++++
 arch/arm/boot/dts/imx7d-sdb-touch.dts      |  9 --------
 arch/arm/boot/dts/imx7d-sdb-touch.dtsi     | 37 ------------------------------
 arch/arm/boot/dts/imx7d-sdb.dts            | 26 +++++++++++++++++++++
 6 files changed, 44 insertions(+), 49 deletions(-)
 delete mode 100644 arch/arm/boot/dts/imx7d-sdb-touch.dts
 delete mode 100644 arch/arm/boot/dts/imx7d-sdb-touch.dtsi

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index aba8f03..4993fac 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -394,7 +394,6 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-sdb-reva-qspi.dtb \
 	imx7d-sdb-reva-touch.dtb \
 	imx7d-sdb-reva-wm8960.dtb \
-	imx7d-sdb-touch.dtb \
 	imx7s-colibri-eval-v3.dtb
 dtb-$(CONFIG_SOC_LS1021A) += \
 	ls1021a-qds.dtb \
diff --git a/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts b/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
index 7c338d5..d3855e8 100644
--- a/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
@@ -6,4 +6,16 @@
  * published by the Free Software Foundation.
  */
 #include "imx7d-sdb-reva.dts"
-#include "imx7d-sdb-touch.dtsi"
+
+&sii902x {
+	status = "disabled";
+};
+
+&ecspi3 {
+	status = "okay";
+
+	tsc2046 {
+		interrupts = <13 0>;
+		pendown-gpio = <&gpio2 13 0>;
+	};
+};
diff --git a/arch/arm/boot/dts/imx7d-sdb-reva.dts b/arch/arm/boot/dts/imx7d-sdb-reva.dts
index 772eb8a..36568bd 100644
--- a/arch/arm/boot/dts/imx7d-sdb-reva.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-reva.dts
@@ -31,6 +31,10 @@
 	};
 };
 
+&ecspi3 {
+	status = "disabled";
+};
+
 &epdc {
 	pinctrl-0 = <&pinctrl_epdc0>;
 	en-gpios = <&gpio_spi 5 GPIO_ACTIVE_LOW>;
diff --git a/arch/arm/boot/dts/imx7d-sdb-touch.dts b/arch/arm/boot/dts/imx7d-sdb-touch.dts
deleted file mode 100644
index 22b3f1a..0000000
--- a/arch/arm/boot/dts/imx7d-sdb-touch.dts
+++ /dev/null
@@ -1,9 +0,0 @@
-/*
- * Copyright (C) 2015 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-#include "imx7d-sdb.dts"
-#include "imx7d-sdb-touch.dtsi"
diff --git a/arch/arm/boot/dts/imx7d-sdb-touch.dtsi b/arch/arm/boot/dts/imx7d-sdb-touch.dtsi
deleted file mode 100644
index 296dee1..0000000
--- a/arch/arm/boot/dts/imx7d-sdb-touch.dtsi
+++ /dev/null
@@ -1,37 +0,0 @@
-/*
- * Copyright (C) 2015 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-&ecspi3 {
-	fsl,spi-num-chipselects = <1>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_ecspi3 &pinctrl_ecspi3_cs>;
-	cs-gpios = <&gpio5 9 0>;
-	status = "okay";
-
-	tsc2046 {
-		compatible = "ti,tsc2046";
-		reg = <0>;
-		spi-max-frequency = <1000000>;
-		pinctrl-names ="default";
-		pinctrl-0 = <&pinctrl_tsc2046_pendown>;
-		interrupt-parent = <&gpio2>;
-		interrupts = <13 0>;
-		pendown-gpio = <&gpio2 13 0>;
-		ti,x-min = /bits/ 16 <0>;
-		ti,x-max = /bits/ 16 <0>;
-		ti,y-min = /bits/ 16 <0>;
-		ti,y-max = /bits/ 16 <0>;
-		ti,pressure-max = /bits/ 16 <0>;
-		ti,x-plat-ohms = /bits/ 16 <400>;
-		linux,wakeup;
-	};
-};
-
-&sii902x {
-	status = "disabled";
-};
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 7e18a9f..a40d766 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -200,6 +200,32 @@
 	status = "okay";
 };
 
+&ecspi3 {
+	fsl,spi-num-chipselects = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi3 &pinctrl_ecspi3_cs>;
+	cs-gpios = <&gpio5 9 0>;
+	status = "okay";
+
+	tsc2046 {
+		compatible = "ti,tsc2046";
+		reg = <0>;
+		spi-max-frequency = <1000000>;
+		pinctrl-names ="default";
+		pinctrl-0 = <&pinctrl_tsc2046_pendown>;
+		interrupt-parent = <&gpio2>;
+		interrupts = <29 0>;
+		pendown-gpio = <&gpio2 29 0>;
+		ti,x-min = /bits/ 16 <0>;
+		ti,x-max = /bits/ 16 <0>;
+		ti,y-min = /bits/ 16 <0>;
+		ti,y-max = /bits/ 16 <0>;
+		ti,pressure-max = /bits/ 16 <0>;
+		ti,x-plat-ohms = /bits/ 16 <400>;
+		linux,wakeup;
+	};
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet1>;
-- 
1.9.1

